apis/Google.Bigquery.V2/Google.Bigquery.V2/BigqueryClient.Queries.cs

```diff
@@ -32,10 +32,22 @@ public virtual BigqueryQueryJob ExecuteQuery(string sql, ExecuteQueryOptions opt
         }
 
         /// <summary>
-        /// Creates a query job, with more facilities than <see cref="ExecuteQuery"/>, including the option
+        /// Executes a command. This overload allows query parameterization, and is preferred over
+        /// <see cref="ExecuteQuery(string, ExecuteQueryOptions)"/> when values need to be passed in.
+        /// </summary>
+        /// <param name="command">The command to execute. Must not be null.</param>
+        /// <param name="options">The options for the operation. May be null, in which case defaults will be supplied.</param>
+        /// <returns>The result of the query.</returns>
+        public virtual BigqueryQueryJob ExecuteQuery(BigqueryCommand command, ExecuteQueryOptions options = null)
```

Comment:
That wouldn't be able to take the optional parameter, and doesn't feel idiomatic in terms of other database clients.
Can add it later if necessary.